bootstrap/src/basis/file.c

```diff
@@ -150,16 +154,29 @@ hm_basis_file_read_complete_inner(value a_bytes, value a_user_data) {
     return caml_copy_int64(res);
 }
 
-// val hm_basis_file_nop_submit_inner: unit >{os}-> uns
+CAMLprim value
+submit_out(hm_opt_error_t oe, hm_user_data_t *user_data) {
+    value a_ret = caml_alloc_tuple(2);
+    Store_field(a_ret, 0, caml_copy_int64((uint64_t)oe));
+    Store_field(a_ret, 1, caml_copy_int64((uint64_t)user_data));
+
+    return a_ret;
+}
+
+// hm_basis_file_nop_submit_inner: unit >{os}-> (int * &Basis.File.Nop.t)
 CAMLprim value
 hm_basis_file_nop_submit_inner(value a_unit) {
-    hm_user_data_t *user_data = hm_ioring_nop_submit(&hm_executor_get()->ioring);
+    hm_opt_error_t oe = HM_OE_NONE;
+
+    hm_user_data_t *user_data = NULL;
+    HM_OE(oe, hm_ioring_nop_submit(&user_data, &hm_executor_get()->ioring));
 
-    return caml_copy_int64((uint64_t)user_data);
+LABEL_OUT:
+    return submit_out(oe, user_data);
 }
 
-// val hm_basis_file_open_submit_inner: Basis.File.Flag.t -> uns -> Stdlib.Bytes.t >{os}->
-//   File.Open.t
+// hm_basis_file_open_submit_inner: Basis.File.Flag.t -> uns -> Stdlib.Bytes.t >{os}->
+//   (int * &File.Open.t)
```

Comment:
FYI, our tuple type syntax is probably going to be `(int, &File.Open.t)`. IIRC we can pull this off (unlike OCaml) due to using postfix type paremetrization (like 1ML).